src/DataTypes/IDataType.h

```diff
@@ -580,6 +581,8 @@ template <> inline constexpr bool IsDataTypeDateOrDateTime<DataTypeDate> = true;
 template <> inline constexpr bool IsDataTypeDateOrDateTime<DataTypeDate32> = true;
 template <> inline constexpr bool IsDataTypeDateOrDateTime<DataTypeDateTime> = true;
 template <> inline constexpr bool IsDataTypeDateOrDateTime<DataTypeDateTime64> = true;
+template <> inline constexpr bool IsDataTypeDate<DataTypeDateTime> = true;
```

Comment:
Looks strange - why it returns that DateTime is Date rather than DateOrDateTime?